spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/client/TraceWebClientBeanPostProcessor.java

```diff
@@ -193,13 +188,16 @@ public void subscribe(CoreSubscriber<? super ClientResponse> subscriber) {
 				log.debug("HttpClientHandler::handleSend: " + span);
 			}
 
+			// NOTE: We are starting the client span for the request here, but it could be
+			// canceled prior to actually being invoked. TraceWebClientSubscription will
+			// abandon this span, if cancel() happens before request().
 			this.next.exchange(wrapper.buildRequest()).subscribe(
-					new WebClientTracerSubscriber(subscriber, context, span, this));
+					new TraceWebClientSubscriber(subscriber, context, span, this));
 		}
 
 	}
 
-	private static final class WebClientTracerSubscriber
+	static final class TraceWebClientSubscriber extends AtomicReference<Span>
```

Comment:
as I notice most of `core` operators implement `Scannable` the same way as `org.springframework.cloud.sleuth.instrument.reactor.ScopePassingSpanSubscriber` maybe this one should also do it?

Reply:
ok sure

Reply:
this implies a good bit of thinking/work so will do it in another change